app/recipe/tests.py

```diff
@@ -0,0 +1,109 @@
+        recipe = Recipe.objects.create(name='Caracoles', description='Caracoles a la llauna o con sofrito')
+        res = self.client.get(recipe_detail_url(recipe.id))
+        serialized_recipe = RecipeSerializer(recipe)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serialized_recipe.data)
+
+    def test_get_recipe_detail_with_ingredients(self):
+        recipe = Recipe.objects.create(name='Gambas al ajillo', description='Gambas con ajo, aceite y perejil')
+        Ingredient.objects.create(name='Gambas', recipe_id=recipe.id)
+        Ingredient.objects.create(name='Ajo', recipe_id=recipe.id)
+        Ingredient.objects.create(name='Aceite', recipe_id=recipe.id)
+
+        serialized_recipe = RecipeSerializer(recipe)
+        response = self.client.get(recipe_detail_url(recipe.id))
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serialized_recipe.data)
+
+    def test_get_recipe_detail_returns_not_found(self):
+        res = self.client.get(recipe_detail_url(1234))
+        self.assertEqual(res.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_create_recipe_with_ingredients(self):
+        ingredientName1 = "Pelota"
+        ingredientName2 = "Caldo"
+        payload = {
+            "name": "Escudella",
+            "description": "Nunca he sabido hacerlo... pero la de Iaia esta bien buena",
+            "ingredients": [{"name": ingredientName1}, {"name": ingredientName2}]
+        }
+
+        response = self.client.post(RECIPE_URL, payload, format='json')
+        recipe = Recipe.objects.get(id=response.data['id'])
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        #Checking everything but the ingredients
+        for key in payload.keys():
+            if key != 'ingredients':
+                self.assertEqual(payload[key], getattr(recipe, key))
+        #Checking ingredients
+        self.assertTrue(recipe.ingredients.get(name=ingredientName1))
+        self.assertTrue(recipe.ingredients.get(name=ingredientName2))
+
+    def test_delete_recipe(self):
+        recipe = Recipe.objects.create(name='Gambas al ajillo', description='Gambas con ajo, aceite y perejil')
+        Ingredient.objects.create(name='Gambas', recipe_id=recipe.id)
+        Ingredient.objects.create(name='Ajo', recipe_id=recipe.id)
+
+        response = self.client.delete(recipe_detail_url(recipe_id=recipe.id))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(Recipe.objects.all().count(), 0)
+
+    def test_update_ingredients(self):
+        recipe = Recipe.objects.create(name='Mikolapiz', description='Helado de vainilla con una fina linea de chocolate')
+        Ingredient.objects.create(name='Nata', recipe_id=recipe.id)
+        Ingredient.objects.create(name='Chocolate', recipe_id=recipe.id)
+
+        ingredientToUpdate = 'Vainilla'
```

Comment:
:laughing:  maybe https://en.wikipedia.org/wiki/Vanilla_Ice is a vain 'nilla
("Vainilla" -> "Vanilla")

Reply:
haha :)